src/umd.intro.js

```diff
@@ -15,34 +15,35 @@
         // Browser globals
         root.Raygun = factory();
     }
-}(this, function () {
+}(this, function() {
 
-  var windw = this || window || global;
-  var originalOnError = windw.onerror;
-  windw.onerror = function (msg, url, line, col, err) {
-    if (originalOnError) {
-      originalOnError(msg, url, line, col, err);
-    }
+    var windw = this || window || global;
+    var originalOnError = windw.onerror;
+    windw.onerror = function(msg, url, line, col, err) {
+        if (originalOnError) {
+            originalOnError(msg, url, line, col, err);
+        }
 
-    if (!err) {
-      err = new Error(msg);
-    }
+        if (!err) {
+            err = new Error(msg);
+        }
 
-    windw['rg4js'].q = windw['rg4js'].q || [];
-    windw['rg4js'].q.push({e: err});
-  };
+        windw['rg4js'].q = windw['rg4js'].q || [];
+        windw['rg4js'].q.push({ e: err });
+    };
+    // Similar approach as the snippet, creates the rg4js proxy function, which is exported in umd.outro.js once the
+    // script is executed, and later overwritten by the loader once it's finished
+    (function(wind) {
+        wind['RaygunObject'] = 'rg4js';
+        wind[wind['RaygunObject']] = wind[wind['RaygunObject']] || function() {
+            if (wind && typeof wind['Raygun'] === 'undefined' ||
+                (typeof document === 'undefined' || document.readyState !== 'complete') || (!wind['RaygunInitialized'])) {
+                // onload hasn't been called, cache the commands just like the snippet
+                (wind[wind['RaygunObject']].o = wind[wind['RaygunObject']].o || []).push(arguments)
```

Comment:
We discussed in person if we should but in some back stop checks

1. Max recursion counter
2. wrapping in a setTimeout()
3. capturing more states (called 0, 1 many times)

However the conclusion was that since this is a hard to replicaster race condition around system functions, me were more likely to add hidden bugs (in unanticipated but legit conditions/uses) as we can prove we would need the back stops.